plugins/arkworks/src/poseidon/config.rs

```diff
@@ -0,0 +1,234 @@
+        Fp(lhs.0 * rhs.0)
+    }
+
+    #[inline]
+    fn add_assign(lhs: &mut Self::Field, rhs: &Self::Field, _: &mut ()) {
+        lhs.0 += rhs.0;
+    }
+
+    #[inline]
+    fn add_const_assign(lhs: &mut Self::Field, rhs: &Self::ParameterField, _: &mut ()) {
+        lhs.0 += rhs.0;
+    }
+
+    #[inline]
+    fn from_parameter(point: Self::ParameterField) -> Self::Field {
+        point
+    }
+}
+
+impl<const ARITY: usize> poseidon::Field<Compiler<Self>> for Spec<ARITY>
+where
+    Self: Specification,
+{
+    type Field = FpVar<<Self as Specification>::Field>;
+
+    #[inline]
+    fn add(lhs: &Self::Field, rhs: &Self::Field, _: &mut Compiler<Self>) -> Self::Field {
+        lhs + rhs
+    }
+
+    #[inline]
+    fn add_const(
+        lhs: &Self::Field,
+        rhs: &Self::ParameterField,
+        _: &mut Compiler<Self>,
+    ) -> Self::Field {
+        lhs + FpVar::Constant(rhs.0)
+    }
+
+    #[inline]
+    fn mul(lhs: &Self::Field, rhs: &Self::Field, _: &mut Compiler<Self>) -> Self::Field {
+        lhs * rhs
+    }
+
+    #[inline]
+    fn mul_const(
+        lhs: &Self::Field,
+        rhs: &Self::ParameterField,
+        _: &mut Compiler<Self>,
+    ) -> Self::Field {
+        lhs * FpVar::Constant(rhs.0)
+    }
+
+    #[inline]
+    fn add_assign(lhs: &mut Self::Field, rhs: &Self::Field, _: &mut Compiler<Self>) {
+        *lhs += rhs;
+    }
+
+    #[inline]
+    fn add_const_assign(lhs: &mut Self::Field, rhs: &Self::ParameterField, _: &mut Compiler<Self>) {
+        *lhs += FpVar::Constant(rhs.0)
+    }
+
+    #[inline]
+    fn from_parameter(point: Self::ParameterField) -> Self::Field {
+        FpVar::Constant(point.0)
+    }
+}
+
+impl<const ARITY: usize> poseidon::Specification for Spec<ARITY>
+where
+    Self: Specification,
+{
+    #[inline]
+    fn apply_sbox(point: &mut Self::Field, _: &mut ()) {
+        point.0 = point.0.pow([Self::SBOX_EXPONENT, 0, 0, 0]);
+    }
+}
+
+impl<const ARITY: usize> poseidon::Specification<Compiler<Self>> for Spec<ARITY>
+where
+    Self: Specification,
+{
+    #[inline]
+    fn apply_sbox(point: &mut Self::Field, _: &mut Compiler<Self>) {
+        *point = point
+            .pow_by_constant([Self::SBOX_EXPONENT])
+            .expect("Exponentiation is not allowed to fail.");
+    }
+}
+
+/// Arity 2 Poseidon Specification
+pub type Spec2 = Spec<2>;
+
+/// Arity 3 Poseidon Specification
+pub type Spec3 = Spec<3>;
+
+/// Arity 4 Poseidon Specification
+pub type Spec4 = Spec<4>;
+
+/// Arity 5 Poseidon Specification
+pub type Spec5 = Spec<5>;
+
+/// Testing Framework
+#[cfg(test)]
+pub mod test {
+    use super::*;
+    use crate::constraint::fp::Fp;
+    use alloc::boxed::Box;
+    use openzl_crypto::{
+        encryption::{Decrypt, Encrypt},
+        poseidon::{
+            encryption::{BlockArray, FixedDuplexer, PlaintextBlock},
+            Constants,
+        },
+    };
+    use openzl_util::rand::{OsRng, Sample};
+
```

Comment:
Maybe add a test checking that the specifications above are indeed secure.

Reply:
Added this to #39